bodhi/models/models.py

```diff
@@ -1649,6 +1649,27 @@ def check_requirements(self, session, settings):
 
         return True, "All checks pass."
 
+    @property
+    def has_negative_karma(self):
+        """Check for negative karma, Returns True if the update has negative karma"""
+        for comment in self.comments:
+            if not comment.anonymous:
+                if comment.karma < 0:
+                    return True
+        return False
+
+    def diasble_autopush_for_critical_update(self, db):
+        """Disable Autopush if a critical update receives any negative karma"""
+        if not self.locked:
+            if self.status is UpdateStatus.testing:
+                if self.autokarma and self.critpath and self.has_negative_karma:
+                    log.info("Disabling Auto Push since the critical update has negative karma")
+                    self.autokarma = False
+                    db.commit()
```

Comment:
I don't think we want to commit the transaction here. In general, we want to commit transactions at a higher level so that any later errors don't cause a partial change to be committed. i.e., if this were committed, and then there were an Exception later in the response handling, we could end up with a "half done" operation. Does that make sense?

We probably just want to make sure there is a db.commit() happening higher in the stack than this, and I suspect there is.